In [3]:
import numpy as np
import pandas as pd
import statsmodels.api as sm

<h2>2) Use the data set BWGHT.DTA that is posted on Canvas and answer the following questions:</h2>

<h3>(a) How many women are in the sample, and how many report smoking during pregnancy (any
amount greater than zero cigarettes)?</h3>

In [4]:
df = pd.read_stata('BWGHT.dta')
df.head()

faminc  cigtax    cigprice  bwght  fatheduc  motheduc  parity  male  white  \
0    13.5    16.5  122.300003    109      12.0      12.0       1     1      1   
1     7.5    16.5  122.300003    133       6.0      12.0       2     1      0   
2     0.5    16.5  122.300003    129       NaN      12.0       2     0      0   
3    15.5    16.5  122.300003    126      12.0      12.0       2     1      0   
4    27.5    16.5  122.300003    134      14.0      12.0       2     1      1   

   cigs    lbwght  bwghtlbs  packs   lfaminc   id  
0     0  4.691348    6.8125    0.0  2.602690  1.0  
1     0  4.890349    8.3125    0.0  2.014903  2.0  
2     0  4.859812    8.0625    0.0 -0.693147  3.0  
3     0  4.836282    7.8750    0.0  2.740840  4.0  
4     0  4.897840    8.3750    0.0  3.314186  5.0

In [19]:
women_num = len(df) # number of women in the sample
df_smokers = df[df['cigs'] > 0] # new dataframe containing women who smoked during pregnancy

print(f'There are {women_num} women in the sample, in which {len(df_smokers)} women smoked during pregnancy') 

There are 1200 women in the sample, in which 184 women smoked during pregnancy


<p>There are 1200 women in the sample, in which 184 women smoked during pregnancy</p>

<h3>(b) What is the average number of cigarettes smoked per day? Is the average a ‘good’measure of the
typical woman in this case? Why?<h3>

In [22]:
df['cigs'].mean() # average cigarettes smoked

2.0416666666666665

The average cigarettes smoked per day is 2.041667 cigarettes per day. This average is not a good measure since it includes women who do not smoke, which is the majority in the sample, in the calculation. This would make the average number of cigarettes smoked per day too few and the study could potentially show inaccurate results.

<h3>(c) Among women who smoked during pregnancy, what is the average number of cigarettes smoked
per day? How does this compare to your answer in part (b), and why?</h3>

In [23]:
df_smokers['cigs'].mean()

13.315217391304348

<p>Among women who smoke during pregnancy, the average number of cigarettes smoked per day is 13.31522 cigarettes per day. This number is a lot larger than the part b)’s average number. The reason is because most women in the sample do not smoke during pregnancy. The number of women who do not smoke during pregnancy is almost seven times larger than women who smoke during the pregnancy. From the dataset, there are 2450 cigarettes smoked per day in total. The average in part b) would be the result of 2450/1200 , while part c) would be the result of 2450/184.</p>

<h3>(d) Find the average of fatheduc in the sample. Why are only 1,033 observations used to compute
this average?</h3>

In [31]:
df_fatheduc = df.dropna(subset=['fatheduc']) # drop a row if a value for 'fatheduc' is missing
print(len(df_fatheduc)) # length of the dataframe after dropping rows with no value for 'fatheduc'

1033


In [33]:
df_fatheduc['fatheduc'].mean() # average value for fatheduc

13.205227492739594

<p>The average of fatheduc in the sample is 13.20523. The reason there are only 1033 samples in the calculation is because there are 167 samples that are missing in fatheduc. In the real world reason, it could be from divorce or death of the father.</p>

<h3>(e) Report the average family income and its standard deviation in dollars?</h3>

In [38]:
print(df['faminc'].mean() * 1000)
print(df['faminc'].std() * 1000)

29471.250534057617
18846.51756286621


<p>The average family income is 29471.25 dollars. The standard deviation is 18846.52 dollars.</p>

***

<h2>3. Let kids denote the number of children ever born to a woman, and let educ denote years of education
for the woman. A simple model relating fertility to years of education is
<br>
<br>
&emsp;kids = β<sub>0</sub> + β<sub>1</sub>educ + u;
<br>
<br>
where u is the unobserved error.</h2>

<h3>(a) What kind of factors are contained in u? Are these likely to be correlated with the level of
education?</h3>

<p>The factors could be something like family income, household location, marriage status, etc. These are likely to be correlated with the level of education. For example, lower number of education years could be the reason for lower family income.</p>

<h3>(b) Will a simple regression analysis uncover the ceteris paribus e¤ect of education on fertility? Explain.</h3>

<p>A simple regression analysis could uncover the ceteris paribus effect of education on fertility assuming E(u) = 0 and/or Δ𝑢 = 0</p>

<h2>4. As noticed from problem 2, the data set BWGHT.DTA contains data on births to women in the
United States. Two of the variables of interest are the dependent variable, infant birth weight (bwght),
and an explanatory variable, average number of cigarettes the mother smoked per day during pregnancy
(cigs). Using the data and what we’ve learned so far, answer the following questions:</h2>

<h3>(a) Obtain estimates for β<sub>0</sub> and β<sub>1</sub> and write down the estimated OLS regression line.</h3>

In [44]:
x = df[['cigs']] # set x variables
y = df[['bwght']] # set y variable
x = sm.add_constant(x) # add constant to the model

In [45]:
reg = sm.OLS(y,x).fit() # fit the model

In [46]:
print(reg.summary()) # print the OLS

                            OLS Regression Results                            
Dep. Variable:                  bwght   R-squared:                       0.022
Model:                            OLS   Adj. R-squared:                  0.021
Method:                 Least Squares   F-statistic:                     26.37
Date:                Wed, 08 Sep 2021   Prob (F-statistic):           3.28e-07
Time:                        17:12:10   Log-Likelihood:                -5295.5
No. Observations:                1200   AIC:                         1.059e+04
Df Residuals:                    1198   BIC:                         1.061e+04
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const        119.6172      0.612    195.436      0.0

<p>𝛽<sub>0</sub> = 119.6172, 𝛽<sub>1</sub> = -0.5149691.
<br>
<br>
The estimated OLS regression line is 
<br>
<br>
𝑏𝑤𝑔ℎ𝑡=119.6172−0.5149691 𝑐𝑖𝑔𝑠</p>

<h3>(b) What is the predicted birth weight when cigs = 0? What about when cigs = 20? Comment on
the di¤erence.</h3>

<p>Predicted birth weight when cigs = 0 → 119.6172 – 0.5149691 (0) = 119.6172 
<br>
Predicted birth weight when cigs = 20 → 119.617 – 0.5149691 (20) = 109.317618</p>

<h3>(c) Does this simple regression necessarily capture a causal relationship between the child’s birth
weight and the mother’s smoking habits? Why or why not?</h3>

<p>This simple regression does not necessarily capture a causal relationship between a child’s birth weight and mother’s smoking habit since R2 is only 0.022, which implies that mother’s smoking habit only explains about 2.2% of the variation in child’s birth weight.</p>

<h3>(d) Given your estimated regression equation in (a), to predict a birth weight of 125 ounces, what
would cigs have to be? Comment.</h3>

<p>cigs have to be -10.4527, which is impossible since people can’t smoke a negative number of cigarettes.</p>

<h3>(e) The proportion of women in the sample who do not smoke while pregnant is about 0:85. Does
this help reconcile your findings from part (d)?</h3>

<p>Apart from using only one factor (cigs) to determine the birth weight of children. It makes sense that the results in d) would be inaccurate since most women in the sample do not smoke during pregnancy. If the sample were to have more women who smoke, there would be more variation which could make the results more accurate.</p>